bitwarden_license/src/Commercial.Core/SecretsManager/Commands/AccessPolicies/DeleteAccessPolicyCommand.cs

```diff
@@ -1,27 +1,135 @@
-﻿using Bit.Core.Exceptions;
+﻿using Bit.Core.Context;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
 using Bit.Core.SecretsManager.Commands.AccessPolicies.Interfaces;
+using Bit.Core.SecretsManager.Entities;
 using Bit.Core.SecretsManager.Repositories;
 
 namespace Bit.Commercial.Core.SecretsManager.Commands.AccessPolicies;
 
 public class DeleteAccessPolicyCommand : IDeleteAccessPolicyCommand
 {
     private readonly IAccessPolicyRepository _accessPolicyRepository;
+    private readonly ICurrentContext _currentContext;
+    private readonly IProjectRepository _projectRepository;
+    private readonly IServiceAccountRepository _serviceAccountRepository;
 
-    public DeleteAccessPolicyCommand(IAccessPolicyRepository accessPolicyRepository)
+    public DeleteAccessPolicyCommand(
+        IAccessPolicyRepository accessPolicyRepository,
+        ICurrentContext currentContext,
+        IProjectRepository projectRepository,
+        IServiceAccountRepository serviceAccountRepository)
     {
+        _projectRepository = projectRepository;
+        _serviceAccountRepository = serviceAccountRepository;
         _accessPolicyRepository = accessPolicyRepository;
+        _currentContext = currentContext;
     }
 
 
-    public async Task DeleteAsync(Guid id)
+    public async Task DeleteAsync(Guid id, Guid userId)
     {
         var accessPolicy = await _accessPolicyRepository.GetByIdAsync(id);
         if (accessPolicy == null)
         {
             throw new NotFoundException();
         }
 
+        switch (accessPolicy)
+        {
+            case UserProjectAccessPolicy ap:
+                if (ap.GrantedProjectId == null)
+                {
+                    throw new BadRequestException();
+                }
+
+                var project = await _projectRepository.GetByIdAsync(ap.GrantedProjectId.Value);
```

Comment:
Any reason we don't include the project in the access policy query?

Reply:
Great suggestion, really like it. 

Resolved in https://github.com/bitwarden/server/pull/2628/commits/cd92ea867e0c629cdebeccbb8a3d9499053cef04